### ISSUE TITLE
Setting Language Disables Other Extensions

### ISSUE THREAD

**Issue by Burtannia** [ndmitchell/hlint#1604]
While configuring my `.hlint.yaml` file for my Yesod project I ran into a strange issue. I decided to dump all my `default-extensions` as arguments into hlint just to ensure correct parsing. I'm using GHC2021 so I also wanted to pass `-XGHC2021`. The minimal example I can give is using template haskell:
```
- arguments: [
  --color,
  --cpp-simple,
  -XGHC2021,
  -XTemplateHaskell,
]
```

This should work just fine to my understanding, in fact `-XTemplateHaskell` is probably redundant because I believe hlint automatically enables it. Either way this produces warnings all over the place of the form:
```
src/Handler/Profile.hs:12:10-31: Suggestion: Redundant bracket
Found:
  do setTitle . toHtml $ userIdent user <> "'s User page"
    $ (widgetFile "profile")
Perhaps:
  do setTitle . toHtml $ userIdent user <> "'s User page"
    $ widgetFile "profile"
```

Removing `-XGHC2021` from the arguments list fixes the issue and the warnings no longer show. My guess is that selecting the language version ignores any other extensions. The issue is also present using `-XHaskell2010`.

As an aside to this, it would be nice if hlint could automatically detect and use the `language` version / `default-extensions` from `package.yaml` to avoid the need for these arguments entirely. This seems to happen via HLS in VSCode but does not happen when running `hlint` manually via the command line.

**Comment by LiamGoodacre** [comment]
Just ran into this too, specifically with GHC2021 & TemplateHaskell; (hlint 3.6.1)

**Comment by LiamGoodacre** [comment]
> to get the GHC2021 extensions and TemplateHaskell then -XGHC201 -XTemplateHaskell should do it

This doesn't appear to work (3.6.1) neither as command line arguments nor arguments in the yaml file.

`Example.hs`
```hs
module Example where

example = $(something)
```

### With command args

```bash
∀ hlint -XGHC2021 -XTemplateHaskell Example.hs
Example.hs:3:11: Error: Parse error: on input `$'
Found:
    module Example where

  > example = $(something)


1 error
```

### With config file

`example.hlint.yaml`
```yaml
- arguments:
    - -XGHC2021
    - -XTemplateHaskell
```

```bash
∀ hlint --hint=example.hlint.yaml Example.hs
Example.hs:3:11: Error: Parse error: on input `$'
Found:
    module Example where

  > example = $(something)


1 error
```

### Only TemplateHaskell (no GHC2021)

```bash
∀ hlint -XTemplateHaskell Example.hs
No hints
```

**Comment by shayne-fletcher** [comment]
quick test here with v3.8 on the command line seems to confirm the reported issue

**Comment by shayne-fletcher** [comment]
i still haven't quite worked out what is up with this but i have a workaround: `hlint -XGHC2021 -XTemplateHaskell -XTemplateHaskellQuotes ...`

**Comment by shayne-fletcher** [comment]
i put a fix up for this in https://github.com/ndmitchell/hlint/pull/1619. bit surprised, i knew that we'd looked at `extensionImplications` before but i couldn't find anywhere we went and used it!

**Comment by LiamGoodacre** [comment]
Thanks a lot for fixing this @shayne-fletcher !
